scripts/render_domain_spine.py

```
@@ -0,0 +1,485 @@
+            result.append('#[repr(' + ', '.join(parts or ['Rust']) + ')]')
+    if item.get('deprecation'):
+        parts = [key + ' = ' + rust_string(value) for key, value in item['deprecation'].items() if value is not None]
+        result.append('#[deprecated' + ('(' + ', '.join(parts) + ')' if parts else '') + ']')
+    return ''.join(value + '\n' for value in result)
+
+
+def format_rust(code):
+    with tempfile.TemporaryDirectory(prefix='domain-spine-rustfmt-') as temporary:
+        path = Path(temporary) / 'declarations.rs'
+        path.write_text(code + '\n')
+        result = subprocess.run([
```

Ruff (0.16.3)

[error] `subprocess` call: check for execution of untrusted input

(S603)